app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -256,13 +267,17 @@ class HomeFragment : Fragment() {
                     //  to some state.
                     recentTabs = getRecentTabs(components),
                     recentHistory = emptyList()
-                ),
+                ).filteredByBlocklist(initialBlocklist),
```

Comment:
@pocmo This is the crux of what I could use some advice on. What's the most idiomatic way to adjust this initial state using this state pattern?